homeassistant/components/mazda/sensor.py

```diff
@@ -0,0 +1,264 @@
+"""Platform for Mazda sensor integration."""
+from homeassistant.const import (
+    CONF_UNIT_SYSTEM_IMPERIAL,
+    LENGTH_KILOMETERS,
+    LENGTH_MILES,
+    PERCENTAGE,
+    PRESSURE_PSI,
+)
+
+from . import MazdaEntity
+from .const import DATA_COORDINATOR, DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the sensor platform."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][DATA_COORDINATOR]
+
+    entities = []
+
+    if coordinator.data is not None:
```

Comment:
This can't happen because you raise a `ConfigEntryNotReady` when the first update request fails.